frontend/app/src/components/home/profile/ViewUserProfile.svelte

```diff
@@ -39,6 +40,8 @@
     $: isPremium = profile?.isPremium ?? false;
     $: diamond = user?.diamond ?? false;
     $: phoneIsVerified = profile?.phoneIsVerified ?? false;
+    $: currentCommunityMembers = client.currentCommunityMembers;
+    $: communityMember = inCommunityContext ? $currentCommunityMembers.get(userId) : undefined;
```

Comment:
Can't this one use `client.getUserDisplayName` now?

Reply:
Yes I had the same thought and modified getUserDisplayName to take into account the notion of "global context".